### PR TITLE
ci: Add support for Karpenter in the kaito pipelines

### PR DESCRIPTION
**Pull request by helayoty**
**Reason for Change**:
- Update e2e pipelines in order to be able to check the gpu-provisioner and Karpenter use cases. This will be handled by adding a new input variable `nodeprovisioner` for the workflow_call. 
- Add make target to install karpenter.
- Unify the variable names used in e2e tests.
- Simplified the e2e pipeline by using the `generate identities` script.
- Use GH variables to pass the k8s version. `AKS_K8S_VERSION`

**Requirements**

- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 
 # Image URL to use all building/pushing image targets
-REGISTRY ?= YOUR_REGISTRY
+REGISTRY ?= hkkarpenter.azurecr.io
```
Comment (Fei-Guo) [inline Makefile:3]: Please clean up these changes for your test with your username in this file.

File: .github/workflows/e2e-workflow.yml [hunk 2]
```diff
@@ -6,6 +6,9 @@ on:
       git_sha:
         type: string
         required: true
+      suite:
```
Comment (Fei-Guo) [inline .github/workflows/e2e-workflow.yml:9]: Please consider renaming variable suite to nodeprovisioner. This improves the readability.

File: .github/workflows/kaito-e2e.yml [hunk 3]
```diff
@@ -17,9 +17,19 @@ permissions:
 
 jobs:
   run-e2e:
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [ gpuprovisioner ]
```
Comment (Fei-Guo) [inline .github/workflows/kaito-e2e.yml:23]: rename the variable name from suite to nodeprovisioner?

**Review by Fei-Guo — APPROVED** [review]

File: .github/workflows/kaito-e2e.yml (hunk 3 above)
Comment (helayoty) [reply .github/workflows/kaito-e2e.yml:23]: Will be fixed on the upcoming PR.

Comment (helayoty) [reply .github/workflows/kaito-e2e.yml:23]: Fixed in #570

**Pull request merged**